WordPress/src/test/java/org/wordpress/android/viewmodel/domains/DomainRegistrationDetailsViewModelTest.kt

```diff
@@ -0,0 +1,654 @@
+    private val shoppingCartRedeemError = RedeemShoppingCartError(PHONE, "Wrong phone number")
+    private val siteChangedError = SiteError(SiteErrorType.GENERIC_ERROR, "Error fetching site")
+    private val domainContactInformationFetchError = DomainContactError(
+            DomainContactErrorType.GENERIC_ERROR,
+            "Error fetching domain contact information"
+    )
+    private val domainSupportedStatesFetchError = DomainSupportedStatesError(
+            DomainSupportedStatesErrorType.GENERIC_ERROR,
+            "Error fetching domain supported states"
+    )
+    private val fetchSupportedCountriesError = FetchSupportedCountriesError(
+            FetchSupportedCountriesErrorType.GENERIC_ERROR,
+            "Error fetching countries"
+    )
+
+    private val createShoppingCartResponse = CreateShoppingCartResponse(
+            siteId.toInt(),
+            cartId,
+            listOf(Product(productId, testDomainName))
+    )
+
+    private val domainProductDetails = DomainProductDetails(productId, testDomainName)
+
+    @Before
+    fun setUp() {
+        site.siteId = siteId
+
+        viewModel = DomainRegistrationDetailsViewModel(dispatcher, store)
+        // Setting up chain of actions
+        setupFetchSupportedCountriesDispatcher(false)
+        setupFetchDomainContactInformationDispatcher(false)
+        setupFetchStatesDispatcher(false)
+        setupCreateShoppingCartDispatcher(false)
+        setupRedeemShoppingCartDispatcher(false)
+        setupFetchSiteDispatcher(false)
+
+        uiStateResults.clear()
+        viewModel.uiState.observeForever { if (it != null) uiStateResults.add(it) }
+        viewModel.domainContactDetails.observeForever(domainContactDetailsObserver)
+        viewModel.showCountryPickerDialog.observeForever(countryPickerDialogObserver)
+        viewModel.showStatePickerDialog.observeForever(statePickerDialogObserver)
+        viewModel.showTos.observeForever(tosLinkObserver)
+        viewModel.handleCompletedDomainRegistration.observeForever(completedDomainRegistrationObserver)
+        viewModel.showErrorMessage.observeForever(errorMessageObserver)
+    }
+
+    @Test
+    fun contactDetailsPreload() = test {
+        viewModel.start(site, domainProductDetails)
+
+        // Verifying that correct actions with expected payloads were dispatched
+        val captor = ArgumentCaptor.forClass(Action::class.java)
+        verify(dispatcher, times(3)).dispatch(captor.capture())
+
+        val actionsDispatched = captor.allValues
+        validateFetchSupportedCountriesAction(actionsDispatched[0])
+        validateFetchDomainContactAction(actionsDispatched[1])
+        validateFetchStatesAction(actionsDispatched[2], primaryCountry.code)
+
+        var preloadStep = 0
+
+        Assertions.assertThat(uiStateResults.size).isEqualTo(5)
+
+        // initial state
+        Assertions.assertThat(uiStateResults[preloadStep].isFormProgressIndicatorVisible).isEqualTo(false)
```

Comment:
I think it would be more readable to have the `uiStateResults[preloadStep]` extracted in a variable.

Reply:
Good idea 👍 I can get rid of the majority of comments that way.